packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -460,26 +546,123 @@ final class SignInStateMachine
     });
   }
 
-  /// Responds to an SRP flow challenge.
+  TotpSetupDetails _createTotpSetupResult(String sharedSecret) =>
+      TotpSetupDetails(
+        sharedSecret: sharedSecret,
+        username: _user.username!,
```

Comment:
Do we need to a guard to check if username is null?

Reply:
Not at this stage, no. It would never be null. This is a case where we know more than the compiler.